### ISSUE TITLE
'\\"' breaks quoting

### ISSUE THREAD

**Issue by Lavender-Tree** [xflr6/graphviz#96]
Write code as this 
```python
from graphviz import Digraph

g = Digraph("CFG")
g.node('0', label='\\" shape=box color=red] # "=12')
g.render()
``` 

You'll get that
```
digraph CFG {
	0 [label="\\" shape=box color=red] # \"=12"]
}
```

**Comment by xflr6** [comment]
I am not sure what the intended outcome is here (it includes a literal backslash in the label, right?), but I suspect this works as documented (see the section on [backslash escapes](https://graphviz.readthedocs.io/en/stable/manual.html#backslash-escapes)).

TLDR; [to be able to make use of Graphviz own backslash escapes](https://nbviewer.jupyter.org/github/xflr6/graphviz/blob/master/examples/graphviz-escapes.ipynb), we need to use raw strings just the same as we do with the `re` module.

I think you might want one of the following two, hope that helps:
```python
In [1]: from graphviz import Digraph
   ...: 
   ...: g = Digraph("CFG")
   ...: g.node('0', label=r'\\', shape='box', color='red')
   ...: g
Out[1]: 
```
![spam](https://user-images.githubusercontent.com/6342379/78503809-36083380-7769-11ea-9baf-5447fe2335b0.png)

```python
In [2]: from graphviz import Digraph
   ...: 
   ...: g = Digraph("CFG")
   ...: g.node('0', label=r'\\" shape=box color=red] # "=12')
   ...: g
Out[2]: 
```
![spam](https://user-images.githubusercontent.com/6342379/78503825-59cb7980-7769-11ea-9dbf-efbf50cf310a.png)

**Comment by Lavender-Tree** [comment]
I want to add a node with '\"%s\"' label (backslash should be outputed)
```python
from graphviz import Digraph

g = Digraph("CFG")

text = '\\"%s\\"'       #  \"%s\"  -> label i want 
text = text.replace('\\', '\\' + '\\') # -> so i need convert to this format?
g.node('0', label=text)
g.render()
```
It's acceptable to do that after realizing the problem. But if  I know little about it (I forget to do a replace), some security problem may be caused.  Like the previous example, I  just set the node's label, but its color also changes.

**Comment by xflr6** [comment]
> I want to add a node with \\"%s\\" label (backslash should be outputed)

Works as intended:

```python
In [1]: from graphviz import Digraph
   ...: 
   ...: g = Digraph("CFG")
   ...: g.node('0', label=r'\\"%s\\"')
   ...: g
Out[1]:
```

![spam](https://user-images.githubusercontent.com/6342379/78536369-08190280-77ee-11ea-9292-9231e9f09efa.png)

> But if I know little about it (I forget to do a replace), some security problem may be caused.

See [the documentation on backslash escapes](https://graphviz.readthedocs.io/en/stable/manual.html#backslash-escapes): To render a string literally (e.g. from an untrusted source) use [graphviz.escape](https://graphviz.readthedocs.io/en/stable/api.html#graphviz.escape) (similar to using `re.escape` for disabling special meaning of backslash-escapes).

**Comment by xflr6** [comment]
Hm, that one can break quoting this way indeed seems unexpected. Need to check if this can be avoided without breaking intended usage of escapes. Might issue a warning.

**Comment by xflr6** [comment]
This required a bit of thought. Users should be able to use all Graphviz escape sequences except for escaping quotes ( `'\\"'`) as escaping of quotes needs to be done by this library. To provide the feature to not produce ill-formed DOT syntax, an odd number of backslashes before a quote needs to be avoided. Different options:

- issue a warning
- raise an error
- replace silently

Assuming there is no use case for interfering with quote escaping, I favour replacement of `'\\"`' with `'\\\\"`, i.e. escaping both the backslash and the quote so that the end result is layouted as `\"`.

```python
In [1]: import graphviz
   ...:  
   ...: g = graphviz.Graph()
   ...: g.node('spam', label='\\" spam')
   ...: print(g)
   ...: g
graph {
	spam [label="\\\" spam"]
}
Out[1]:
```

![spam](https://user-images.githubusercontent.com/6342379/78607667-d04aa300-785f-11ea-80a4-4df88a7f6847.png)

Can you check if [this branch](https://github.com/xflr6/graphviz/compare/quoting) would work for you (basically users could still use all Graphviz escapes unless you wrap their input in `graphviz.escape`, but it would not be possible to break the syntax this way, I think)?

**Comment by Lavender-Tree** [comment]
```python
from graphviz import Digraph

g = Digraph("CFG")

text = '\\"%s\\"'       # `\"%s\"` -> label i want 
g.node('0', label=text) # it works 
text = '\\\\"%s\\"'
g.node('1', label=text) # but ...

g.render()
```

![image](https://user-images.githubusercontent.com/44033552/78628445-b55f4980-78c6-11ea-8688-25312ebc6f4f.png)

**Comment by xflr6** [comment]
> g.node('1', label=text) # but ...

Not sure what you mean, can you be more specific?

AFAICT, the example you give works exactly as [proposed above](https://github.com/xflr6/graphviz/issues/96#issuecomment-610053007). For illustration:

```
In [1]: import graphviz
   ...: 
   ...: for n in range(9):
   ...:     x = '\\' * n + '"'
   ...:     y = graphviz.lang.quote(x)
   ...:     print(f'{x:9s} -> {y}')
"         -> "\""
\"        -> "\\\""
\\"       -> "\\\""
\\\"      -> "\\\\\""
\\\\"     -> "\\\\\""
\\\\\"    -> "\\\\\\\""
\\\\\\"   -> "\\\\\\\""
\\\\\\\"  -> "\\\\\\\\\""
\\\\\\\\" -> "\\\\\\\\\""
```

The other option would be to raise an error on an odd number of backslashes followed by a quote.

**Comment by xflr6** [comment]
I guess a third option would be to ignore the odd-numbered backslash, so `'"'` and `'\\"'` would be the same (no backslash), as well as `'\\"'` and  `'\\\"'` (single backslash) and so on.

**Comment by xflr6** [comment]
I think the option to ignore escaping of a quote by the user is better here, i.e. to treat both `'"'`and `'\\"'` the same viz. a literal quote, see [quoting](https://github.com/xflr6/graphviz/compare/quoting) branch.

```
In [1]: import graphviz
   ...:  
   ...: for n in range(6):
   ...:     x = '\\' * n + '"'
   ...:     y = graphviz.lang.quote(x)
   ...:     print(f'{x:9} -> {y}')
"         -> "\""
\"        -> "\""
\\"       -> "\\\""
\\\"      -> "\\\""
\\\\"     -> "\\\\\""
\\\\\"    -> "\\\\\""
```

**Comment by xflr6** [comment]
fixed in [0.14 release](https://pypi.org/project/graphviz/0.14/#files) (529d3d25c934c8502dea2166b26f155a73d5a7f5)
